frontend/catalyst/pennylane_extensions.py

```diff
@@ -1419,7 +1567,7 @@ def __call__(self):
         if mode == EvaluationMode.QUANTUM_COMPILATION:
             return self._call_with_quantum_ctx(ctx)
         elif mode == EvaluationMode.CLASSICAL_COMPILATION:
-            return self._call_with_classical_ctx()
+            return self._call_with_classical_ctx(ctx)
```

Comment:
Why do we need to add the context when classical? In cond I don't think you actually use it

Reply:
I think I actually use it [here](https://github.com/PennyLaneAI/catalyst/pull/370/files/4478f648015e4fbc8c6b4cd359e7c96cb2e892a4#diff-8754bab66a7845aa8e14eb1ab898192548b9ab878d595f57b9c6639930e2c6bfR1538), in the `trace_function` call. This `ctx` is our context which keeps lower-level Jax tracing stack objects. This PR attempts to switch our code to a more uniform way of handling the nested tracing, thus the new state variable. In the previous version we in relied on Jax modifying its contexts without letting us know, see [here](https://github.com/google/jax/blob/88a60b808c1f91260cc9e75b9aa2508aae5bc9f9/jax/_src/interpreters/partial_eval.py#L2189).